develop/tutorials/articles/25-frontend-customizations/xx-converting-struts-action-wrappers-to-mvc-commands.markdown

```diff
@@ -1,60 +1,99 @@
+
+As a review, legacy StrutsActionWrappers needed to implement all the methods,
+`processAction`, `render`, and `serveResource`, even if only one method was 
+being customized. Each of these methods can now be customized independently, 
+using different classes, making the logic simpler and easier to maintain. 
+Depending on which method you customized in your StrutsActionWrapper, you will 
+need to use the matching MVCCommand shown below:
 
 -   processAction ->  MVCActionCommand
 -   render -> MVCRenderCommand
 -   serveResource -> MVCResourceCommand
 
-Once you know which MVCCommand you will need to know which is the
-associated mapping that is used by that MVCCommand.
+<!--
+
+Can you show the code for this, using the beginning example? What action would
+the login example use, and what are some other examples using the other actions?
+
+-->
 
-For most of the cases this will be a trivial task because it’s the same
-value defined in the legacy struts-action. In the first example, the
-mapping was  /login/login and this value remains the same in Liferay 7.
-However, this might have changed in some of the portlets, so it’s better
-to double check it.
+Next you'll need to determine the associated mapping that is used by the 
+MVCCommand.
 
-This mapping can be found in Liferay source code. Depending on the url
-it will be a different param:
+For most cases, the MCVCCommand mapping is the same mapping defined in the 
+legacy struts-action.
 
--   RenderURLs will contain a parameter named “mvcRenderCommandName”
--   ActionURLs will be contained in the attribute “name“ of the taglib
-    or with the parameter ActionRequest.ACTION_NAME
--   RenderURLs will be contained as the attribute “id”.
+Using the beginning login example once again, the `struts-action-path` mapping, 
+`/login/login`, remains the same for the MVCCommand mapping in Liferay 7.
+
+Some of the portlets mapping may have changed, so it’s best to check Liferay's 
+source code to determine the correct mapping.
+
+Depending on the URL it will be a different param:
+
+-   RenderURLs will contain a parameter named `mvcRenderCommandName`
+-   ActionURLs will be contained in the attribute `name` of the taglib
+    or with the parameter `ActionRequest.ACTION_NAME`
+-   RenderURLs will be contained as the attribute `id`.<!-- Is this suppose to be the RenderURLs again, or something else? -->
+
+<!--
+
+Can you show the code for this, using the beginning example? In which cases
+would you use each param?
+
+-->
 
 Once you have this information you should be able to override the
 MVCCommand following these instructions:
 
+<!--
+
+What would code up to this point look like? What should I have at this point to
+move onto the tutorials shown below? Is this all done in a `liferay-hook.xml`?
+
+I'm guessing not. So what file are these code changes being made in?
+
+A legacy code example and a finished converted code(using the same example)
+would be good to have here.
+
+Thanks!
+
+-->
```

Comment:
All the information about how the code would look like and how to create mvc commands is here https://dev.liferay.com/develop/tutorials/-/knowledge_base/7-0/overriding-mvc-commands